test/cmdlineTests/standard_optimizer_yulDetails_optimiserSteps_empty_sequence_whitespaces_metada/output.json

```diff
@@ -0,0 +1,19 @@
+{
+    "contracts":
+    {
+        "A":
+        {
+            "C":
+            {
+                "metadata": "{\"compiler\":{\"version\":\"<VERSION REMOVED>\"},\"language\":\"Solidity\",\"output\":{\"abi\":[],\"devdoc\":{\"kind\":\"dev\",\"methods\":{},\"version\":1},\"userdoc\":{\"kind\":\"user\",\"methods\":{},\"version\":1}},\"settings\":{\"compilationTarget\":{\"A\":\"C\"},\"evmVersion\":\"cancun\",\"libraries\":{},\"metadata\":{\"bytecodeHash\":\"ipfs\"},\"optimizer\":{\"details\":{\"constantOptimizer\":false,\"cse\":false,\"deduplicate\":false,\"inliner\":false,\"jumpdestRemover\":true,\"orderLiterals\":false,\"peephole\":true,\"simpleCounterForLoopUncheckedIncrement\":true,\"yul\":false,\"yulDetails\":{\"optimizerSteps\":\":\"}},\"runs\":200},\"remappings\":[]},\"sources\":{\"A\":{\"keccak256\":\"0xb63d3c030f127aecaa7a1e223e41ef58ce5b4404939d397ccfd9ca0eb11d757b\",\"license\":\"GPL-3.0\",\"urls\":[\"bzz-raw://567242af52de5d6870167f16a7f46d7bf27fb2b101cf99fd9ab48c645dfe7ecd\",\"dweb:/ipfs/QmamT3HqCmEbVvWwQiYSfUZRn5XwHRRkEe29BpbEJadxDT\"]}},\"version\":1}"
```

Comment:
I noticed here that the metadata was not preserved. The `optimizerSteps` value is just the colon instead of a whitespace before it and two whitespaces after it.
The cause is 
https://github.com/ethereum/solidity/blob/ae9bcabd954c20e9610a3f3fe64e24cbcf153375/libsolidity/interface/CompilerStack.cpp#L1732
being defined if the sequence is the empty optimizer sequence, although we accept whitespaces and linebreaks.

Reply:
I fixed that in https://github.com/ethereum/solidity/pull/14967/commits/a455829160c29c730212d9634568faedc2a9eb13.

Reply:
True. But changing it so that this is preserved also technically changes the metadata so I'd keep it out of this PR.

Like I said in https://github.com/ethereum/solidity/pull/14967#discussion_r1551741993, this is fine to do, but I'd rather do any such changes explicitly, in a separate PR. 

Also, I think that the right way to make it consistent is to be always stripping whitespace from the sequence, means we'd leave this bit as is and instead change the rest to match :)

Reply:
Ok, I will separate this commit apart.